src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -488,58 +479,62 @@ public static function getCategoryId(string $title, string $language = null): in
-             INNER JOIN meta AS m ON p.meta_id = m.id
-             WHERE i.id = ? AND p.status = ?
-             LIMIT 1',
-            [(int) $id, 'active']
-        );
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
+            ->getRepository(Comment::class);
+        $comment = $repository->find($id);
+
+        if ($comment instanceof Comment) {
+            $commentData = $comment->toArray();
+
+            // I know this is dirty, but as we don't have full entities yet we
+            // need to fetch the post separately and inject it into the comment
+            $postData = (array)BackendModel::getContainer()->get('database')
+                ->getRecord(
+                    'SELECT p.id AS post_id, p.title AS post_title, m.url AS post_url
+                     FROM blog_posts AS p
+                     INNER JOIN meta AS m ON p.meta_id = m.id
+                     WHERE p.id = ? AND p.status = ? AND p.language = ?
+                     LIMIT 1',
+                    [
+                        (int)$comment->getId(),
+                        'active',
+                        $comment->getLocale()->getLocale(),
+                    ]
+                );
+
+            return array_merge(
+                $commentData,
+                $postData
+            );
+        }
+
+        return [];
     }
 
-    /**
-     * Get multiple comments at once
-     *
-     * @param array $ids The id(s) of the comment(s).
-     *
-     * @return array
-     */
     public static function getComments(array $ids): array
     {
-        return (array) BackendModel::getContainer()->get('database')->getRecords(
-            'SELECT *
-             FROM blog_comments AS i
-             WHERE i.id IN (' . implode(', ', array_fill(0, count($ids), '?')) . ')',
-            $ids
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
+                                  ->getRepository(Comment::class);
+        return array_map(
+            function(Comment $comment) {
+                $commentData = $comment->toArray();
+                // I really don't know why this is returned in a non UNIX-timestamp format
+                $commentData['created_on'] = $comment->getCreatedOn()->format('Y-m-d H:i:s');
+
+                return $commentData;
+            },
+            $repository->findById($ids)
         );
     }
 
-    /**
-     * Get a count per comment
-     *
-     * @return array
-     */
     public static function getCommentStatusCount(): array
     {
-        return (array) BackendModel::getContainer()->get('database')->getPairs(
-            'SELECT i.status, COUNT(i.id)
-             FROM blog_comments AS i
-             WHERE i.language = ?
-             GROUP BY i.status',
-            [BL::getWorkingLanguage()]
-        );
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
```

Comment:
idem

Reply:
Fixed in cd9cef81ef540cf410bc31750966a00591aa06ae